tests/lib/rules/no-identical-names.js

```diff
@@ -0,0 +1,111 @@
+    RuleTester = require("eslint").RuleTester;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+
+ruleTester.run("no-identical-title", rule, {
+
+    valid: [
+        [
+            "module(\"module\");",
+            "test(\"test\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"module1\");",
+            "test(\"it1\", function() {});",
+            "test(\"it2\", function() {});"
+        ].join("\n"),
+        [
+            "test(\"it1\", function() {});",
+            "test(\"it2\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"title\", function() {});",
+            "test(\"title\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"module1\");",
+            "test(\"it1\", function() {});",
+            "module(\"module2\");",
+            "test(\"it1\", function() {});"
+        ].join("\n"),
+        [
+            "module(\"module1\");",
+            "module(\"module2\");"
+        ].join("\n"),
+        [
+            "test(\"test\" + n, function() {});",
+            "test(\"test\" + n, function() {});"
+        ].join("\n"),
+        {
+            code: [
+                "test(`it${n}`, function() {});",
+                "test(`it${n}`, function() {});"
+            ].join("\n"),
+            env: {
+                es6: true
+            }
+        }
+    ],
+
+    invalid: [
+        {
+            code: [
+                "module(\"module1\");",
+                "test(\"it1\", function() {});",
+                "test(\"it1\", function() {});"
+            ].join("\n"),
+            errors: [{
+                message: "Test name is used multiple times in the same module.",
+                column: 6,
+                line: 3
+            }]
+        },
+        {
+            code: [
+                "test(\"it1\", function() {});",
+                "test(\"it1\", function() {});"
+            ].join("\n"),
+            errors: [{
+                message: "Test name is used multiple times in the same module.",
+                column: 6,
+                line: 2
+            }]
+        },
+        {
+            code: [
+                "module(\"module1\");",
+                "module(\"module1\");"
+            ].join("\n"),
+            errors: [{
+                message: "Module name is used multiple times.",
+                column: 8,
+                line: 2
+            }]
+        },
+        {
+            code: [
+                "module(\"module1\");",
+                "test(\"it\", function() {});",
+                "module(\"module1\");"
+            ].join("\n"),
+            errors: [{
+                message: "Module name is used multiple times.",
+                column: 8,
+                line: 3
+            }]
+        }
```

Comment:
General note: Could we have a few tests using modules with function expressions (without nesting)?

For example, here's a valid case:

```js
QUnit.module("module1", function() {
    QUnit.test("test1", function() {});
    QUnit.test("test2", function() {});
});
```

I'm fairly sure they'll work, but it will be good to know with certainty. :smile: